plugins/tech-insights-backend/src/service/fact/factRetrievers/catalogFactRetriever.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  FactRetriever,
+  FactRetrieverContext,
+} from '@backstage/plugin-tech-insights-node';
+import { CatalogClient } from '@backstage/catalog-client';
+import { Entity } from '@backstage/catalog-model';
+import isEmpty from 'lodash/isEmpty';
+import camelCase from 'lodash/camelCase';
+import { get } from 'lodash';
+
+type Options = {
+  annotations?: string[];
+};
+
+export const createCatalogFactRetriever = (
+  { annotations = [] }: Options = { annotations: [] },
+): FactRetriever => ({
+  id: 'entityRetriever',
```

Comment:
This potentially needs an entity filter to avoid running these on certain entities e.g. users and groups (as some of these facts won't make sense for these).